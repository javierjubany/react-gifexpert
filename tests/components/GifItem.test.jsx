
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test of GifItem', () => {

    const title = 'Test Title';
    const url = 'http://test.url/';


    test('snapshot is correct', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el URL y ALT indicado', () => {

        render(<GifItem title={title} url={url} />);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })


    test('debe mostrar el titulo en el componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    })


})