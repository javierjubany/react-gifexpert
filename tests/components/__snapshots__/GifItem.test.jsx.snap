// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifItem snapshot is correct 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="http://test.url/"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
